### PR TITLE
Introduce ktLint

### PR DESCRIPTION
**Pull request by hannesa2**
You can run int on CI with `./gradlew ktLintCheck`
The fine tuning in file `.editorconfig` is up to you. I added some super annoying exception of strict rules, adding more or less is up to you.

Btw, in current state you have to fix some files.

Library: https://github.com/owncloud/android-library/pull/390

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[*]
+max_line_length = 150
+
+[*.{kt, kts}]
+disabled_rules=no-consecutive-blank-lines,no-wildcard-imports,import-ordering,max-line-length,import-ordering,no-blank-line-before-rbrace,final-newline,indent,no-multi-spaces,comment-spacing,parameter-list-wrapping
```
Comment (hannesa2) [inline .editorconfig:5]: eg, if you want to check for a laste empty line, you have to remove `final-newline`

**Comment by jesmrec** [comment]
We will check which rules are the best to enable.

**Comment by hannesa2** [comment]
Currently all checks are fulfilled.
To be honest, it's more efficient to merge current state, add check to your CI. The longer you'll wait the more violation you will generate.
To be more tolerant or to have more strict new rules can be done anytimes

**Comment by abelgardep** [comment]
I kinda agree with you. Let's move this forward, we will discuss which rules are interesting for the project in the short term.

**Review by abelgardep — APPROVED** [review]
LGTM

**Comment by jesmrec** [comment]
Nothing to check from my side here

**Comment by abelgardep** [comment]
Merged, thanks @hannesa2 for the contribution 👍

**Pull request merged**